heron/schedulers/src/java/com/twitter/heron/scheduler/marathon/MarathonScheduler.java

```diff
@@ -169,19 +199,19 @@ protected ArrayNode getPorts(ObjectMapper mapper) {
 
     for (String portName : MarathonConstants.PORT_NAMES) {
       ObjectNode port = mapper.createObjectNode();
-      port.put(MarathonConstants.PORT, 0);
+      port.put(MarathonConstants.DOCKER_CONTAINER_PORT, 0);
       port.put(MarathonConstants.PROTOCOL, MarathonConstants.TCP);
-      port.put(MarathonConstants.PORT_NAME, portName);
+      port.put(MarathonConstants.HOST_PORT, 0);
```

Comment:
portName isn't important for it to run within Marathon, but setting the HOST_PORT to 0 allows the host to randomly choose a port on the machine to use. So it does need to be in there.